setup.sh

```diff
@@ -1,29 +1,47 @@
 #!/bin/sh
+# Check if nix is already installed
+if ! which nix > /dev/null 2>&1; then
+  # Install Nix
+  curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | \
+    sh -s -- install --no-confirm --extra-conf "trusted-users = $(whoami)"
 
-# Install Nix
-curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | \
-  sh -s -- install --no-confirm --extra-conf "trusted-users = $(whoami)"
-
-# Source nix configuration
-. /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
+  # Source nix configuration
+  . /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
+fi
 
 # Run `om health`
 echo "\n# Check nix health"
 nix --accept-flake-config run github:juspay/omnix health
 
-# Setup nixos-unified-template
-echo "\n# Setting up home-manager & direnv"
-nix --accept-flake-config run github:juspay/omnix -- \
-  init github:juspay/nixos-unified-template#home -o ~/.config/home-manager \
-  --non-interactive \
-  --params '{"username":"'$(id -un)'", "git-name":"'$(id -un)'", "git-email":"'$(id -un)'@juspay.in"}'
+health_out=$(nix --accept-flake-config run github:juspay/omnix -- health --json 2>/dev/null)
+is_nix_healthy=$?
+
+echo $health_out | nix run nixpkgs#jq -- -e '.info.nix_installer.type == "DetSys"' > /dev/null
+is_detsys_used=$?
+
+if [ $is_nix_healthy -ne 0 ] && [ $is_detsys_used -ne 0 ]; then
+  echo "\n# Nix unhealthy"
+  echo "\n# Uninstall Nix: <https://nixos.asia/en/howto/uninstall-nix>. Post uninstall, re-run the script."
+  exit 1
+fi
```

Comment:
The code here is not self-explanatory as to why, for example, `om health` being unhealthy (say, trusted-users check failed or Nix version is old) must have the user uninstall Nix.

Reply:
I have added a comment now, also added a TODO to re-evaluate if suggesting Uninstall is too harsh.

Reply:
I was thinking if it is better to suggest uninstall only if rosetta check fails?